### PR TITLE
[flutter_tools] fix response format of flutterVersion, flutterMemoryInfo

### LINKED ISSUE #54755: regression wrt flutter_tools use of the vm service protocol
We're seeing a few, serious, regressions wrt flutter tools, flutter run, and the vm service protocol. I bisected the issues to:

> commit 4851888864348524b4065e18cd6e2597f906fd8d
> [flutter_tools] Migrate to package:vm_service 4: trigonometric boogaloo (#54132)

The issues are:
- failures in our (DevTools) CI: https://travis-ci.org/github/flutter/devtools/jobs/672707154; 'Service has disappeared (112) from s0.flutterVersion()'
- after a flutter run, flutter tools can lose the connection to the running app, and exit back to the cli, even though the app continues to run
- when quitting on the command line (`q`), the app in the ios simulator continues to run
- DevTools tries to call the `flutterVersion` service extension registered by flutter_tools. However, we get an error back from the VM, as after registering the service extension, flutter tools then looses its connection to the VM (the service extension provider is no longer available). This causes other issues in the DevTools UI, including the hot reload and restart actions not being available.

I can repro this all locally and can test any fix. These issues are serious enough that we should fix as soon as possible or roll back the change.

cc @jonahwilliams

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

When the response structure of this JSON is incorrect, it causes the client to crash. Fix the format and write an integration test that confirms registration and calling works correctly.

Fixes https://github.com/flutter/flutter/issues/54755

### CODE REVIEW

File: packages/flutter_tools/lib/src/vmservice.dart [hunk 1]
```diff
@@ -255,7 +255,12 @@ class VMService implements vm_service.VmService {
       final Map<String, Object> versionJson = version.toJson();
       versionJson['frameworkRevisionShort'] = version.frameworkRevisionShort;
       versionJson['engineRevisionShort'] = version.engineRevisionShort;
-      return versionJson;
+      return <String, dynamic>{
+        'result': <String, Object>{
+          'type': 'Success',
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/vmservice.dart:260]: This looks correct (comparing to the `compileExpression` implementation).

cc @bkonyi - it looks like each client registering a vm service extension needs to be very careful about the response it creates

**Review by devoncarew — APPROVED** [review]
lgtm

@bkonyi, it would be worth thinking about ways we could improve either the VM service protocol or the VM service lib so that service protocol extensions could just return their results (instead of carefully packing their results into well-formed data structures).

**Comment by Hixie** [comment]
Might be worth seeing if there's something we can do to abstract this out and make it easier to get it right -- e.g. having a general API for sending responses so that only one place needs to create this JSON blob. (As a separate PR, obviously, not trying to hold this up.)

**Comment by jonahwilliams** [comment]
Yeah, Ideally the callback here would accept a vm_service.Response type. Then we could construct a well-typed object like vm_service.Success and avoid trying to match up the JSON structure

**Pull request merged**